### PR TITLE
BCF-3225 - Implement forwarder fallback if forwarder not present as a transmitter on OCR2 aggregator

### PR DESCRIPTION
**Pull request by ilija42**
- Pass OCR2 aggregator address to forwarders to validate forwarders being present on transmitters list

### CODE REVIEW

File: core/chains/evm/forwarders/forwarder_manager.go [hunk 1]
```diff
@@ -131,6 +134,42 @@ func (f *FwdMgr) ForwarderFor(addr common.Address) (forwarder common.Address, er
...
+	}
+
+	transmitters, err := offchainAggregator.GetTransmitters(&bind.CallOpts{Context: f.ctx})
+	if err != nil {
+		return common.Address{}, errors.Errorf("failed to get ocr2 aggregator transmitters: %s", err.Error())
+	}
+
+	for _, fwdr := range fwdrs {
+		if !slices.Contains(transmitters, fwdr.Address) {
+			f.logger.Criticalw("Forwarder is not set as a transmitter", "forwarder", fwdr.Address, "ocr2Aggregator", ocr2Aggregator, "err", err)
```
Comment (george-dorin) [inline core/chains/evm/forwarders/forwarder_manager.go:154]: Shouldn't this be a `info`? `critical` seems a bit too much

Comment (ilija42) [reply core/chains/evm/forwarders/forwarder_manager.go:154]: I don't think that this should ever happen

Comment (george-dorin) [reply core/chains/evm/forwarders/forwarder_manager.go:154]: I thought that if the forwarder is not first in the set then the log would be emitted, even tho the next forwarder might be the right one

Comment (ilija42) [reply core/chains/evm/forwarders/forwarder_manager.go:154]: I'm not sure if we ever want to have forwarders on the node which aren't configured for every OCR2 contract. Might be wrong, I'll double check

Comment (ilija42) [reply core/chains/evm/forwarders/forwarder_manager.go:154]: Yeah, double checked, this shouldn't ever happen outside of the migration

File: core/services/ocr2/delegate.go [hunk 2]
```diff
@@ -496,7 +496,7 @@ func GetEVMEffectiveTransmitterID(jb *job.Job, chain legacyevm.Chain, lggr logge
 		if chain == nil {
 			return "", fmt.Errorf("job forwarding requires non-nil chain")
 		}
-		effectiveTransmitterID, err := chain.TxManager().GetForwarderForEOA(common.HexToAddress(spec.TransmitterID.String))
+		effectiveTransmitterID, err := chain.TxManager().GetForwarderForEOAOCR2(common.HexToAddress(spec.TransmitterID.String), common.HexToAddress(spec.ContractID))
```
Comment (krehermann) [inline core/services/ocr2/delegate.go]: is the existing GetForwarderForEOA method still used somewhere else?

Comment (ilija42) [reply core/services/ocr2/delegate.go]: Yeah,  OCR1, keepers and eth tx task

Comment (ilija42) [reply core/services/ocr2/delegate.go]: I won't touch keepers for now as I am not 100% sure if it makes sense there and eth tx task shouldn't use this and will get deprecated anyway

Comment (ilija42) [reply core/services/ocr2/delegate.go]: Also trying to minimize the scope as this affects the migration

File: common/txmgr/txmgr.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
```
Comment (jmank88) [inline common/txmgr/txmgr.go]: I think geth is not meant to be used from the `common` subtree.

File: common/txmgr/txmgr.go [hunk 4]
```diff
@@ -649,6 +660,10 @@ func (n *NullTxManager[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) Cre
 func (n *NullTxManager[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) GetForwarderForEOA(addr ADDR) (fwdr ADDR, err error) {
 	return fwdr, err
 }
+func (n *NullTxManager[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) GetForwarderForEOAOCR2(eoa common.Address, ocr2AggregatorID common.Address) (fwdr ADDR, err error) {
```
Comment (jmank88) [inline common/txmgr/txmgr.go]: Should this use the generic type?
```suggestion
func (n *NullTxManager[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) GetForwarderForEOAOCR2(eoa common.Address, ocr2AggregatorID ADDR) (fwdr ADDR, err error) {
```

File: core/chains/evm/forwarders/forwarder_manager.go [hunk 5]
```diff
@@ -2,13 +2,16 @@ package forwarders
...
 import (
 	"context"
+	"slices"
 	"sync"
 	"time"
 
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/graph-gophers/graphql-go/errors"
```
Comment (jmank88) [inline core/chains/evm/forwarders/forwarder_manager.go]: Is this an auto-import mistake?
```suggestion
	"errors"
```

File: core/chains/evm/forwarders/forwarder_manager_test.go [hunk 6]
```diff
@@ -2,19 +2,22 @@ package forwarders_test
...
 	"testing"
 	"time"
 
-	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
-
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core"
+	"github.com/smartcontractkit/libocr/gethwrappers2/testocr2aggregator"
```
Comment (jmank88) [inline core/chains/evm/forwarders/forwarder_manager_test.go]: Unsorted imports

**Review by george-dorin — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**